src/main/python/hydra/lib/runtestbase.py

```diff
@@ -221,6 +222,74 @@ def create_hydra_app(self, name, app_path, app_args, cpus, mem, ports=None, cons
         self.refresh_app_info(name)
         return r
 
+    def create_hydra_app_group(self, name, app_path, app_args, cpus, mem, apps_in_group,
+                               total_apps, ports=None, constraints=None):
+        """
+        Create an application in groups
+        @args:
+        name:                                      name of the app
+        app_path:                                  path to launch app from
+        app_args:                                  app args
+        cpus:                                      required cpus
+        mem:                                       required  mem
+        apps_in_group:                             number of apps to group together
+        total_apps:                                total apps to launch
+        ports:                                     port list (optional)
+        constaints:                                constaints optional
+        """
+        assert(name not in self.apps)
+        r = self.__mt.create_app(
+            name, MarathonApp(cmd=self.get_cmd(app_path, app_args),
+                              cpus=cpus, mem=mem,
+                              ports=ports,
+                              constraints=constraints,
+                              uris=[self.get_app_uri()]))
+        self.apps[name] = {'app': r, 'type': 'script'}
+        self.wait_app_ready(name, 1)
+        self.scale_and_verify_app(name, total_apps)
+        self.create_app_groups(name, apps_in_group)
+        return r
+
+    def create_app_groups(self, name, apps_in_group):
+        """
+        Create relevant dictionaries containting info about
+        process info categorized into groups.
+        @args:
+        name:            Name of the app
+        apps_in_group:   Number of apps to group together
+
+        NOTE: This only groups process info like ip:port to talk to that process
+              it DOES NOT group process launches
+        """
+        l.info("Grouping process port info")
+        self.all_launched_ip_port_map = {}
+        assert(name in self.apps)
+        if name not in self.app_group:
+            self.app_group[name] = {}
+        tasks = self.get_app_tasks(name)
+        for task in tasks:
+            app_ip = self.get_ip_hostname(task.host)
+            for app_rep_port in task.ports:
+                self.all_launched_ip_port_map[task.id + '_PORT' + str(app_rep_port)] = \
```

Comment:
The existing data structure doesnt fit in here. Since it is tied to [app_id] = {portmap, stats etc}

I need to create a new one for grouping info, for that I need the full map or re write the existing data strucure more. Was getting complex